request.go

```diff
@@ -226,14 +210,11 @@ func (r *Requester) Do(ar *APIRequest, responseStruct interface{}, options ...in
 		}
 	}
 
-	if r.BasicAuth != nil {
-		req.SetBasicAuth(r.BasicAuth.Username, r.BasicAuth.Password)
-	}
-
 	for k := range ar.Headers {
 		req.Header.Add(k, ar.Headers.Get(k))
 	}
 
+	req.SetBasicAuth(r.BasicAuth.Username, r.BasicAuth.Password)
```

Comment:
That was something I was trying to figure out through re-testing.  I know that it still passes the tests, so maybe there's a test that needs to be added to verify that it's in a known-good-state so it can be used.